src/python/pants/build_graph/target.py

```diff
@@ -794,12 +760,44 @@ def strict_dependencies(self, dep_context):
     """
     strict_deps = self._cached_strict_dependencies_map.get(dep_context, None)
     if strict_deps is None:
+      closure_kwargs = dep_context.target_closure_kwargs
+      respect_intransitive = closure_kwargs.get('respect_intransitive')
+      include_scopes = closure_kwargs.get('include_scopes')
+      exclude_scopes = closure_kwargs.get('exclude_scopes')
+      def pred(target, dep):
+        if type(target) in dep_context.alias_types:
+          return True
+        if respect_intransitive and not dep.transitive:
```

Comment:
I _think_ that intransitive should be applied on a source node, rather than on the dest? Perhaps not.

Reply:
Intransitive in the original level_predicate operates on the dest. This just continues that behavior. I think that the current implementation is slightly wrong because it will return `False` even if the source target is the root of the traversal.